src/Socket/messages-send.ts

```diff
@@ -940,6 +949,32 @@ export const makeMessagesSocket = (config: SocketConfig) => {
 				logger.debug({ jid }, 'adding device identity')
 			}
 
+			if (!isNewsletter) {
+				if (
+					reportingMessage &&
+					shouldIncludeReportingToken(reportingMessage) &&
+					reportingMessage.messageContextInfo?.messageSecret &&
+					msgId
```

Comment:
I don't think we need this `msgId` here; it always exists.

One question: in which cases should we NOT send the reporting token? 

Seems like we are not sending on retryResponses neither newsletter. Is that right?